pkgs/top-level/all-packages.nix

```diff
@@ -14104,23 +14104,17 @@ let
 
   cuyo = callPackage ../games/cuyo { };
 
-  dfhack = callPackage_i686 ../games/dfhack {
-    inherit (pkgsi686Linux.perlPackages) XMLLibXML XMLLibXSLT;
-  };
-
   dhewm3 = callPackage ../games/dhewm3 {};
 
   drumkv1 = callPackage ../applications/audio/drumkv1 { };
 
-  dwarf_fortress = callPackage_i686 ../games/dwarf-fortress {
-    SDL_image = pkgsi686Linux.SDL_image.override {
-      libpng = pkgsi686Linux.libpng12;
-    };
-    inherit (pkgsi686Linux.perlPackages) XMLLibXML XMLLibXSLT;
+  dwarf-fortress-packages = callPackage ../games/dwarf-fortress { };
+
+  dwarf-fortress = dwarf-fortress-packages.dwarf-fortress.override {
     enableDFHack = config.dwarfFortress.enableDFHack or false;
```

Comment:
Ah, I wasn't aware of that policy. I think I was emulating the `chromium` and `wine` package when I added those config options. Might I ask the rationale behind the policy? Is it documented somewhere? I rather liked the config since it makes it plain how the package can be configured. It's nice to just look in `all-packages.nix` and see the possible config items right there.

That said, I'd rather not break a policy just because I don't agree with it. Let's go with option 1.